### PR TITLE
Replace generated cleaner code with concrete implementation

### PR DESCRIPTION
**Pull request by robbavey**
No need to to perform codegen to create JDK specific compatible classes to
call the ByteBuffer cleaner, as we are dropping support for Java 8.

Note:

Cannot be committed until release manager process stops needing java 8 (https://github.com/elastic/infra/pull/33275)

### CODE REVIEW

**Comment by robbavey** [comment]
Acceptance Tests: https://logstash-ci.elastic.co/job/elastic+logstash+master+multijob-acceptance/335/
Docker acceptance tests: https://logstash-ci.elastic.co/job/elastic+logstash+master+multijob-docker-acceptance/197/

File: logstash-core/src/main/java/org/logstash/LogstashJavaCompat.java [hunk 1]
```diff
@@ -17,80 +17,13 @@
...
 package org.logstash;
 
-import java.io.StringReader;
-import java.util.ArrayList;
-import java.util.Collection;
-import org.codehaus.janino.ClassBodyEvaluator;
-import org.logstash.ackedqueue.io.ByteBufferCleaner;
-
 /**
  * Logic around ensuring compatibility with Java 8 and 9 simultaneously.
```
Comment (andsel) [inline logstash-core/src/main/java/org/logstash/LogstashJavaCompat.java:23]: This comment no longer makes sense, maybe the class itself has a misleading name in this context.
The only remaining method is `isJavaAtLeast` which is used only in `org.logstash.instrument.monitors.ProcessMonitor.getCpuLoad()` [coderef](https://github.com/elastic/logstash/blob/0c2645c9c6e181334b85f8414165d4e36c9671a0/logstash-core/src/main/java/org/logstash/instrument/monitors/ProcessMonitor.java#L122)

**Review by andsel — COMMENTED** [review]
The changes looks good to me, I've left a note on the effectiveness of the class naming and existence.

**Review by andsel — APPROVED** [review]
LGTM

**Comment by andsel** [comment]
Jenkins test this please

**Comment by robbavey** [comment]
@FireBurn From `8.0` onwards, Java 11 or Java 17 is required to build and run Logstash - please try again, and let us know if you have any further issues

**Comment by robbavey** [comment]
@FireBurn Thanks for pointing that out - the `README` in our `main` branch mentioned that JDK11 was required to build, but was not backported to other branches. I've put up a PR that also mentions JDK17 support, and I'll backport that to the relevant branches once the initial PR is approved.

> Does build.grade need to change:
>
> project.sourceCompatibility = JavaVersion.VERSION_1_8
> project.targetCompatibility = JavaVersion.VERSION_1_8
> 

> [PR](https://github.com/elastic/logstash/pull/13657) in progress

**Pull request merged**